src/libs/Navigation/linkTo.ts

```diff
@@ -181,12 +179,11 @@ export default function linkTo(navigation: NavigationContainerRef<RootStackParam
         } else if (type === CONST.NAVIGATION.TYPE.UP) {
             action.type = CONST.NAVIGATION.ACTION_TYPE.REPLACE;
 
-            // If this action is navigating to the ModalNavigator and the last route on the root navigator is not already opened ModalNavigator then push
-        } else if (isModalNavigator(action.payload.name) && !isTargetNavigatorOnTop) {
+            // If this action is navigating to ModalNavigator or FullScreenNavigator and the last route on the root navigator is not already opened Navigator then push
+        } else if ((action.payload.name === NAVIGATORS.FULL_SCREEN_NAVIGATOR || isModalNavigator(action.payload.name)) && !isTargetNavigatorOnTop) {
```

Comment:
Hi, after this change, I can't open the workspace page on native and mWeb.

https://github.com/Expensify/App/assets/50919443/d0595066-e0a2-4186-9f81-070ed21f702b

Reply:
Hi! Thanks for noticing that. I've noticed that 2 days ago this PR was merged https://github.com/Expensify/App/pull/38977. It has introduced some changes to the navigation logic. When I was working on this PR, these changes weren't merged yet. I'll try to adjust my fix :)

Reply:
Came here to point out the same issue